userland/test/test_server.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <userland_msgs/AddTwoInts.h>
+
+void add(
+  const std::shared_ptr<userland_msgs::AddTwoInts::Request> req,
+  std::shared_ptr<userland_msgs::AddTwoInts::Response> res)
+{
+  std::cout << "Got request #" << req->a << std::endl;
```

Comment:
I just created a ticket and submitted two PR to address this.

https://github.com/ros2/rclcpp/issues/21
https://github.com/ros2/rclcpp/pull/22
https://github.com/ros2/examples/pull/15